From b8444c5857432101b58cba43d43906b903da77ff Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Sun, 25 Sep 2016 21:36:56 +0200
Subject: [PATCH 1/2] Recorder includes to prevent conflict with vsnprintf

Some files #include <dshow.h>
This is a C header which also #include's stdio.h which adds a #define vsnprintf
This #define vsnprint conflicts with QtCore/qstring.h so reorder the includes
a bit to prevent this situation
---
 src/multimedia/video/qvideosurfaceformat.h       | 1 +
 src/plugins/directshow/common/directshowglobal.h | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/multimedia/video/qvideosurfaceformat.h b/src/multimedia/video/qvideosurfaceformat.h
index 83785ac0..25607d08 100644
--- a/src/multimedia/video/qvideosurfaceformat.h
+++ b/src/multimedia/video/qvideosurfaceformat.h
@@ -44,6 +44,7 @@
 #include <QtCore/qpair.h>
 #include <QtCore/qshareddata.h>
 #include <QtCore/qsize.h>
+#include <QtCore/qstring.h>
 #include <QtGui/qimage.h>
 #include <QtMultimedia/qvideoframe.h>
 
diff --git a/src/plugins/directshow/common/directshowglobal.h b/src/plugins/directshow/common/directshowglobal.h
index 5f391710..3ae3b8aa 100644
--- a/src/plugins/directshow/common/directshowglobal.h
+++ b/src/plugins/directshow/common/directshowglobal.h
@@ -43,6 +43,7 @@
 #include <dshow.h>
 
 #include <QtCore/qglobal.h>
+#include <QtCore/qstring.h>
 #include <QtCore/qloggingcategory.h>
 
 QT_BEGIN_NAMESPACE
-- 
2.15.1

